### PR TITLE
only show lint messages of the current linting file

### PR DESCRIPTION
**Pull request by liigo**
I'm working on Rust Linter, which would like to support Rust's package manager, Cargo. Cargo was designed to build a package (or project), which is a group of many source files. While linting a file inside a package, we build the full package (use Cargo). (You can't just compile only one of these files individually.) There are maybe some errors, cause by files other than the current linting one, which need to be ignored. This PR changes the behavior to only show lint messages of the current linting file.

This is a backward compatible improvement.

### CODE REVIEW

File: lib/linter.coffee [hunk 1]
```diff
@@ -254,7 +260,12 @@ class Linter
   #        (optional)
   #   colStart: column to on which to start a higlight (optional)
   #   colEnd: column to end highlight (optional)
+  #   file: the file that a message was caused by (optional)
   computeRange: (match) ->
+    if match.file?
+      if match.file isnt @lintingFile
+        # TODO: show a message to indicate there are errors in other files
```
Comment (hd-deman) [inline lib/linter.coffee]: did you mean show something in treeView?

Comment (liigo) [inline lib/linter.coffee]: @hd-deman Perhaps. I know little about Linter itself. Maybe it could be improved in another PR later?

**Comment by reem** [comment]
Any news here? This is a big hurdle to using atom with rust.

**Comment by iam4x** [comment]
@hd-deman Can you take care of this one?

**Comment by HitomiTenshi** [comment]
Any news on the merge situation? I'd like to use rust with atom.

**Comment by bb010g** [comment]
bumpity bump

**Comment by HitomiTenshi** [comment]
bumping intensifies

**Comment by bb010g** [comment]
Woo!

**Comment by AsaAyers** [comment]
To be clear this wasn't merged. `linter` has been completely rewritten so all existing issues were closed at once from #571 

I believe this can be solved in `linter-rust` (or whatever) by migrating to the [new API](https://github.com/AtomLinter/Linter/wiki/Linter-API). All linters will be required to migrate to the new API, but the timeline hasn't been set yet.

**Comment by liigo** [comment]
To be clear, I'd like to say, I didn't submit soooooooo many commits. I don't known what happens here.

@AsaAyers I'll try new Linter api if possible. Thank you!

**Pull request closed without merging**